addons/web/static/tests/views/fields/reference_field_tests.js

```diff
@@ -964,6 +966,71 @@ QUnit.module("Fields", (hooks) => {
         }
     );
 
+    QUnit.test(
+        "Change model_field value on ReferenceField with model_field option and select invalid value (tree list in form view)",
+        async function (assert) {
+            serverData.models.turtle.records[0].partner_ids = [1];
+            serverData.models.partner.records[0].reference = "product,41";
+            serverData.models.partner.records[0].model_id = 20;
+
+            let updateCount = 0;
+            patchWithCleanup(ReferenceField.prototype, {
+                setup() {
+                    this._super();
+                    onWillRender(() => {
+                        updateCount++;
+                        if (updateCount > 100) {
+                            // we need to throw an error directly for the test to fail
+                            throw Error("ReferenceField has entered an infinite render loop");
```

Comment:
What happens if you don't have this? Does the test still crash on runbot?

If so I would remove it and simply assert that everything went correctly in your test case.